src/big_core_input.cpp

```diff
@@ -159,26 +170,31 @@ void BigCoreInput::getTile(T* data, uint64_t imageNum, uint64_t tileNum)
 	if (tileNum >= this->numberOfTiles)
 		throw "Number of tiles out of bound!";
 
-	size_t dataSize = this->imageWidth * this->imageHeight * this->numberOfPlanes;
-    // RV: zde je prave ta alokace dat, ktera je trochu problem, protoze uzivatel pak pravdepodobne zapomene uklidit
-    // RV: radeji tedy verzi, kde uzivatel uz musi predat alokovane pole, do ktereho se data jen zapisi
-	// data = new T[dataSize];
-
-	// RV: tady to bohuzel nebude tak jednoduche. Toto jsem prave myslel tou "slozitou" metodou. Musi se vyzobat jednotlive elementy a poskladat do pole
-	//size_t ptr = 0;
-	//if (!this->isUniformDataType)
-	//{
-	//	for (size_t i = 0; i < imageNum; i++)
-	//	{
-	//		ptr += this->imageSize(this->dataType[i]);
-	//	}
-	//}
-	//else
-	//	ptr = this->imageSize(this->dataType[0]) * imageNum;
+	// first do some preparation
+	uint64_t iterators[] = { 0, 0, 0 };
+	uint64_t stoppers[3];
+	size_t index = 0;
+	for (size_t i = 0; i < 5; i++)
+	{
+		if (this->dataOrder[i] != 1 && this->dataOrder[i] != 2)
+		{
+			stoppers[index] = this->permutation[i];
+			index++;
+		}
+	}
```

Comment:
this places iterators in correct order and in the 3-for cycles tile can be correctly iterated